### PR TITLE
[BEAM-1345]  Add internal comments to metrics

### PR DESCRIPTION
**Pull request by aaltay**
R: @robertwb @chamikaramj 
cc: @pabloem

### CODE REVIEW

File: sdks/python/apache_beam/metrics/metric.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
 from apache_beam.metrics.metricbase import MetricName
 
 
+__all__ = ['Metrics', 'MetricsFilter']
```
Comment (robertwb) [inline sdks/python/apache_beam/metrics/metric.py]: MetricsFilter is Experimental in Java, should be marked as such below.

File: sdks/python/apache_beam/metrics/execution.py [hunk 2]
```diff
@@ -34,11 +34,12 @@
 
 from apache_beam.metrics.cells import CounterCell, DistributionCell
 
+__all__ = [
+    'MetricKey', 'MetricResult', 'MetricsEnvironment', 'MetricsContainer']
```
Comment (robertwb) [inline sdks/python/apache_beam/metrics/execution.py]: Should these all be public? Likely not any of them?

Comment (aaltay) [inline sdks/python/apache_beam/metrics/execution.py]: Maybe `MetricResult` and `MetricKey` only?

Comment (aaltay) [inline sdks/python/apache_beam/metrics/execution.py]: They are not used, and the file already has an internal comment, I will revert this.

**Comment by aaltay** [comment]
PTAL.

**Comment by robertwb** [comment]
LGTM

**Pull request closed without merging**